spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/DefaultMqttPahoClientFactory.java

```diff
@@ -95,9 +105,270 @@ public void setConnectionOptions(MqttConnectOptions options) {
 
 	@Override
 	public MqttConnectOptions getConnectionOptions() {
-		return this.options;
+		if (this.serverUris == null) {
+			return this.options;
+		}
+		else {
+			return new ExtendedOptions();
+		}
 	}
 
+	@Override
+	public void setServerUris(List<String> serverUris) {
+		this.serverUris = serverUris;
+	}
+
+	private class ExtendedOptions extends MqttConnectOptions {
+
+		@Override
+		public char[] getPassword() {
+			return DefaultMqttPahoClientFactory.this.options.getPassword();
+		}
+
+		@Override
+		public void setPassword(char[] password) {
+			DefaultMqttPahoClientFactory.this.options.setPassword(password);
```

Comment:
M-m-m. May be this delegate should be just "read-only"?
And all the setters should throw `UnsupportedOperationException` instead of delegation?